client/data/newsletter-categories/use-newsletter-categories-query.tsx

```diff
@@ -0,0 +1,58 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type NewsletterCategoryQueryProps = {
+	blogId: number;
+};
+
+type NewsletterCategoryQueryResponse = {
+	newsletter_categories: NewsletterCategoryResponse[];
+};
```

Comment:
These names may be a bit confusing `NewsletterCategoryQueryResponse` isn't related to the query but to the request, and `NewsletterCategoryResponse` is just a part of the response shape. Just an idea, but maybe you could have just one type with the whole response shape, something like this:

```
type NewsletterCategoriesResponse = {
	newsletter_categories: {
		term_id: number;
		name: string;
		description: string;
		order: number;
	}[];
};
```

Reply:
BTW I did a few changes on the response, you can check the latest version here D117699-code

Reply:
Oh, nice! That makes the js version of the response and the snake case version the same shame for NewsletterCategory type. Great 👍